tests/queries/0_stateless/03230_alter_with_mixed_mutations_and_remove_materialized.sql

```diff
@@ -0,0 +1,13 @@
+DROP TABLE IF EXISTS a SYNC;
+CREATE TABLE a (x String, y String MATERIALIZED 'str') ENGINE = ReplicatedMergeTree('/clickhouse/{database}/a', 'r1') ORDER BY x;
```

Comment:
The bug doesn't reproduce with RMT, but this test will fail with SMT. To reproduce the issue SMT is fundamental, as no other engines parse pure metadata alters.